### PR TITLE
Added vfunc to IconView

### PR DESCRIPTION
**Pull request by hfiguiere**
This is a follow up on #931 - vfuncs were missing.

### CODE REVIEW

File: src/subclass/icon_view.rs [hunk 1]
```diff
@@ -1,13 +1,252 @@
...
 
 use super::container::ContainerImpl;
 use ContainerClass;
+use IconView;
 use IconViewClass;
+use MovementStep;
+use TreePath;
+
+pub trait IconViewImpl: IconViewImplExt + ContainerImpl + 'static {
+    fn item_activated(&self, icon_view: &IconView, path: &mut TreePath) {
```
Comment (sdroege) [inline src/subclass/icon_view.rs]: This should probably not be a mutable reference, or is this supposed to be changeable by the subclass? It maps to a signal and to a normal method, the former doesn't mention anything about it being modifiable, the latter takes an immutable reference.

**Comment by sdroege** [comment]
Looks good to me otherwise

File: src/subclass/icon_view.rs (hunk 1 above)
Comment (hfiguiere) [reply src/subclass/icon_view.rs]: I had no choice. The underlying vfunc in the class expect a `*mut` and it seems it's the only way.

`    pub item_activated: Option<unsafe extern "C" fn(*mut GtkIconView, *mut GtkTreePath)>,`

Comment (EPashkin) [reply src/subclass/icon_view.rs]: Note that it `&IconView` even it `*mut GtkIconView` 😉 
So better check docs and possible just remove `mut`

Comment (sdroege) [reply src/subclass/icon_view.rs]: `glib::translate::mut_override()` to go from `*const T` to `*mut T`. That's generally caused by C API not using `const` correctly, so all pointers are `*mut` on the Rust side.

Comment (EPashkin) [reply src/subclass/icon_view.rs]: @sdroege Strange, but seems this function is only one that not initialized by default even if gtk4
https://github.com/GNOME/gtk/blob/mainline/gtk/gtkiconview.c#L369
https://github.com/GNOME/gtk/blob/mainline/gtk/gtkiconviewprivate.h#L55

Comment (sdroege) [reply src/subclass/icon_view.rs]: Yeah, just doing nothing if not implemented by a subclass.

Comment (hfiguiere) [reply src/subclass/icon_view.rs]: ok I use `mut_override()` here. And removed mut

File: src/subclass/icon_view.rs [hunk 2]
```diff
@@ -1,13 +1,255 @@
...
+unsafe extern "C" fn icon_view_item_activated<T: ObjectSubclass>(
+    ptr: *mut gtk_sys::GtkIconView,
+    path: *mut gtk_sys::GtkTreePath,
+) where
+    T: IconViewImpl,
+{
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap = from_glib_borrow(ptr);
+    let path = from_glib_full(path);
```
Comment (sdroege) [inline src/subclass/icon_view.rs]:
```suggestion
    let path = from_glib_borrow(path);
```

Comment (hfiguiere) [reply src/subclass/icon_view.rs]: done

**Comment by sdroege** [comment]
@GuillaumeGomez Please merge :)

**Comment by GuillaumeGomez** [comment]
Thanks!

**Pull request merged**